### PR TITLE
Deprecate "debug" and remove "--debug" / "--debug-brk" flags

### PR DESCRIPTION
**Pull request by juergba**
### Description

#3885 is required to be merged first.
The support of Node v6 will be dropped shortly.

The Node argument `debug` and the flags`--debug`/`--debug-brk` are deprecated / invalid in Node v8.0.0 or newer. There is no need to keep alive Mocha/Node hybrid argument/flags, users should make use of `inspect`, `--inspect` and `--inspect-brk` instead.

### Description of change

- deprecate `debug` argument, but still pass it over to Node, which prints a deprecation warning and matches `debug` into `inspect`.
- remove `--debug` / `--debug-brk` flags, which are invalid in Node >= v8.0.0
- conflicts between `inspect`, `debug`, `--inspect` and `--inspect-brk` are now handled by Node directly.

### Applicable issues

#3884 (partially)

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -1174,15 +1174,15 @@ Use the _inverse_ of the match specified by `--grep` or `fgrep`.
 
 Requires either `--grep` or `--fgrep` (but not both).
 
-### `--debug, --inspect, --debug-brk, --inspect-brk, debug, inspect`
+### `--inspect, --inspect-brk, inspect`
 
-> _BREAKING CHANGE in v6.0.0; `-d` is no longer an alias for `--debug`._ > _Other updates in v6.0.0:_ > _In versions of Node.js implementing `--inspect` and `--inspect-brk`, `--debug` and `--debug-brk` are respectively aliases for these two options._ > _Likewise, `debug` (not `--debug`) is an alias for `inspect` (not `--inspect`) in Node.js versions where `debug` is deprecated._
+> \_BREAKING CHANGE in v7.0.0; `--debug`, `--debug-brk` and `debug` are deprecated.
```
Comment (outsideris) [inline docs/index.md]: deprecated or removed?
IMO, it is removed. If it is deprecated, it should print deprecation message.

**Comment by juergba** [comment]
@outsideris thanks.

I installed Node v8.0.0 on my laptop and got following results:

`node --debug`:  => **node stops running** and exits with 
```
(node:4476) [DEP0062] DeprecationWarning: `node --debug` and `node --debug-brk` are invalid. Please use `node --inspect` or `node --inspect-brk` instead.
```

`node --debug-brk`:  => **node stops running** and exits with 
```
(node:9188) [DEP0062] DeprecationWarning: `node --debug` and `node --debug-brk` are invalid. Please use `node --inspect` or `node --inspect-brk` instead.
```

`node debug`:  => **node continues running** with
```
(node:8252) [DEP0068] DeprecationWarning: `node debug` is deprecated. Please use `node inspect` instead.
< Debugger listening on ws://127.0.0.1:9229/d843ca77-e169-410a-a1e5-adafca7d2b04
< For help see https://nodejs.org/en/docs/inspector
```

**Comment by juergba** [comment]
So my suggestion is:
- `--debug`, `--debug-brk`: we remove them from Mocha and change the docu accordingly. So they don't get deprecated, but removed from Mocha.
- `debug`: we just deprecate this flag (not remove) and pass it over to Node. Node will then print its deprecation warning and transform it to `inspect`.

@outsideris can you live with this proposition?

**Comment by juergba** [comment]
Please note: `node-environment-flags`, the [polyfill](https://github.com/boneskull/node-environment-flags/blob/master/flags.json#L111) for `process.allowedNodeEnvironmentFlags` returns `false` for both flags `--debug` and `--debug-brk` when Node >=8.0.0.

**Comment by juergba** [comment]
@craigtaub @outsideris @boneskull 
I will merge this PR within the next few days.

**Pull request merged**